components/fan/speed.rst

```diff
@@ -30,15 +30,7 @@ Configuration variables:
   :ref:`output <output>` to use for the oscillation state of this fan. Default is empty.
 - **direction_output** (*Optional*, :ref:`config-id`): The id of the
   :ref:`output <output>` to use for the direction state of the fan. Default is empty.
-- **speed** (*Optional*): Set the float values for each speed setting:
-
-  - **low** (**Required**, float): Set the value for the low speed
-    setting. Must be in range 0 to 1. Defaults to 0.33.
-  - **medium** (**Required**, float): Set the value for the medium speed
-    setting. Must be in range 0 to 1. Defaults to 0.66.
-  - **high** (**Required**, float): Set the value for the high speed
-    setting. Must be in range 0 to 1. Defaults to 1.
-
+- **speed_levels** (*Optional*): Set the number of supported speed levels. Defaults to 100.
```

Comment:
```suggestion
- **speed_levels** (*Optional*, int): Set the number of supported discrete speed levels. Each step will
  represent a calculated percentage value to the output of the fan. E.g. ``2`` will send 50% and
  100% while ``100`` will allow 1% increments in the output. Defaults to ``100``.
```

Reply:
This one is a bit confusing, but I know what you are aiming at. I'll try to rephrase it later after work.